src/main/index.ts

```diff
@@ -18,30 +20,20 @@ const renderWindow = async () => {
   });
 
   // Depending on the environment the frontend will either load from the react server or the static html file
-  if (isDev) {
-    mainWindow.loadURL('http://localhost:3000/');
-  } else {
-    mainWindow.loadFile('./build/index.html');
-  }
+  mainWindow.loadURL(isDev ? 'http://localhost:3000/' : './build/index.html');
 
-  // Detect if devtools was somehow opened outside development
+  // Detect if devtools was somehow opened outside development and close it
   mainWindow.webContents.on('devtools-opened', () => {
-    if (!isDev) {
-      mainWindow.webContents.closeDevTools();
-    }
+    if (!isDev) mainWindow.webContents.closeDevTools();
   });
+  
 };
 
+// Render the main window on app open
 app.on('ready', renderWindow);
 
-app.on('window-all-closed', () => {
-  if (process.platform !== 'darwin') {
-    app.quit();
-  }
-});
-
-app.on('activate', () => {
-  if (mainWindow === null) {
-    renderWindow();
-  }
-});
+// Additional check to close app on macos
+app.on('window-all-closed', () => !isMacOS && app.quit());

```

Comment:
Avoid one liners in templates to make them more accessible to beginners who require these templates to learn.